types/filter.d.ts

```diff
@@ -10,11 +10,16 @@ export function filter<A, P extends A>(
   <B extends A>(list: readonly B[]): P[];
 };
 
-// filter(() => boolean)
+// filter(() => boolean, list)
```

Comment:
`// filter(() => boolean)` was correct as is. More accurately it's `// filter(() => boolean)(list | dict)`